spec/controllers/ems_infra_controller_spec.rb

```diff
@@ -299,8 +299,8 @@
         .to receive(:register_and_configure_nodes).and_return("SUCCESS")
       post :register_nodes, :params => {:id => @ems.id, :nodes_json => @nodes_example, :register => 1}
       expect(controller.send(:flash_errors?)).to be_falsey
-      expect(response.body).to include("redirected")
-      expect(response.body).to include("ems_infra")
+      expect(response.status).to eq(302)
+      assert_redirected_to(%r(#{controller.ems_infra_path(@ems.id)}))
```

Comment:
I use a regex here as the I didn't think it was necessary to test the whole URL, such as:

```
http://test.host/ems_infra/1040?display=hosts&flash_msg=Nodes+were+added+successfully.+Refresh+queued.
```